File: mobile/android/app/src/main/java/ru/edu/asu/minobrlabs/sensors/common/AbstractWorker.java
package ru.edu.asu.minobrlabs.sensors.common;

public abstract class AbstractWorker extends Thread implements Runnable {
    protected volatile boolean running;

    public AbstractWorker() {
        this.running = false;
    }

    @Override
    public abstract void run();

    @Override
    public void start() {
        if (!running) {
            running = true;
            super.start();
        }
    }

    public void kill() {
        running = false;
    }
}

File: desktop/windows/MinobrLabsProject/MinobrLabsProject/sensors/common/AbstractWorker.cs
using System.Threading;

namespace MinobrLabsProject.sensors.common
{
    public abstract class AbstractWorker
    {
        protected volatile bool running;
        private Thread thread;

        public AbstractWorker()
        {
            running = false;
            thread = new Thread(new ThreadStart(run));
            thread.SetApartmentState(ApartmentState.STA);
            thread.IsBackground = true;
        }
        
        public abstract void run();
        
        public void start()
        {
            if (!running)
            {
                running = true;
                thread.Start();
            }
        }

        public void kill()
        {
            running = false;
            thread.Abort();
        }
    }
}
